### PR TITLE
feat: text component

### PR DESCRIPTION
**Pull request by vincentaudebert**
## Type

- Feature

#### What is expected?

New Text component

#### The following changes where made:

(Describe what you did)

1. Adds Text component to SWUI

## Relevant logs and/or screenshots

Screenshot 2022-06-06 at 12 31 41

### CODE REVIEW

**Comment by vincentaudebert** [comment]
Known issue: tooltip doesn't trigger on first hover

Test coverage to improve + visual testing

**Review by matthprost — CHANGES_REQUESTED** [review]
Typography should be marked as deprecated with a description into storybook and should have `@deprecated` comment to mark it in JS/TS too

File: src/components/Text/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import recursivelyGetChildrenString from '../../helpers/recursivelyGetChildrenString'
+import { Color, typography } from '../../theme'
+import capitalize from '../../utils/capitalize'
+import Tooltip from '../Tooltip'
+
+export const PROMINENCES = {
+  default: '',
+  strong: 'strong',
+  weak: 'weak',
+}
```
Comment (matthprost) [inline src/components/Text/index.tsx:13]: There is also stronger as prominence but only for neutral variant

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Typography/index.tsx [hunk 2]
```diff
@@ -315,6 +315,9 @@ const TextWithTooltip = ({
   )
 }
 
+/**
+ * @deprecated Has been replaced by component Text, please this one instead.
```
Comment (matthprost) [inline src/components/Typography/index.tsx:319]:
```suggestion
 * @deprecated Has been replaced by component Text, please use this one instead.
```

**Review by matthprost — APPROVED** [review]

**Review by DorianMaliszewski — APPROVED** [review]

**Pull request merged**